### PR TITLE
Use Symbol for uplo in Bidiagonal constructors

### PR DESCRIPTION
**Pull request by fredrikekre**
- Consistent with `Symmetric`/`Hermitian` wrappers
- It is clearer to use `:U`/`:L` instead of `true`/`false`
- The `Char` constructor was originally for internal use only; remove and remove docstring

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -136,6 +136,9 @@ Deprecated or removed
   * The method `replace(s::AbstractString, pat, r, count)` with `count <= 0` is deprecated
     in favor of `replace(s::AbstractString, pat, r, typemax(Int))` ([#22325]).
 
+  * `Bidiagonal` constructors have been re-factored to use a `Symbol` for the upper/lower
```
Comment (KristofferC) [inline NEWS.md]: "have been re-factored to" -> "now"?

**Comment by fredrikekre** [comment]
AV is the libgit2 failure.

**Comment by fredrikekre** [comment]
Rebased.

**Review by KristofferC — APPROVED** [review]

File: base/linalg/bidiag.jl [hunk 2]
```diff
@@ -4,121 +4,74 @@
...
  7  2  ⋅  ⋅
  ⋅  8  3  ⋅
  ⋅  ⋅  9  4
 ```
 """
-Bidiagonal(dv::AbstractVector{T}, ev::AbstractVector{T}, isupper::Bool) where {T} = Bidiagonal{T}(collect(dv), collect(ev), isupper)
+function Bidiagonal(dv::AbstractVector{T}, ev::AbstractVector{T}, uplo::Symbol) where T
+    Bidiagonal{T}(collect(dv), collect(ev), char_uplo(uplo))
+end
 Bidiagonal(dv::AbstractVector, ev::AbstractVector) = throw(ArgumentError("did you want an upper or lower Bidiagonal? Try again with an additional true (upper) or false (lower) argument."))
```
Comment (fredrikekre) [inline base/linalg/bidiag.jl]: Woops, missed to update this one. Can we just remove this method? IMO a `MethodError` is better:

```julia
julia> Bidiagonal(a, b)
ERROR: MethodError: no method matching Bidiagonal(::Array{Float64,1}, ::Array{Float64,1})
Closest candidates are:
  Bidiagonal(::Array{T,1}, ::Array{T,1}, ::Char) where T at linalg/bidiag.jl:15
  Bidiagonal(::AbstractArray{T,1}, ::AbstractArray{T,1}, ::Symbol) where T at linalg/bidiag.jl:59
  Bidiagonal(::AbstractArray{Td,1}, ::AbstractArray{Te,1}, ::Symbol) where {Td, Te} at linalg/bidiag.jl:64
```

**Pull request merged**